### DISCUSSION TITLE
Browser support

### DISCUSSION THREAD

**Discussion by bumi** [BetaHuhn/metadata-scraper#147] [Ideas]
Currently it is not possible to run in the browser because of the `got` dependency. 
If that dependency could be removed it could be possible to run in the browser context. 

got could either be replaced by a different library that also support the browser context or the dependency for a http library could be removed completely and left up to the user to get the html. (e.g. by using got)

**Comment by bumi** [comment]
Never mind... domino would also need to be replaced (which would work, but I assume this is not the goal of the project)

**Comment by BetaHuhn** [comment]
I never needed to something like `metadata-scraper` in the browser, that's why it's currently built to only run on the server.

For what use case would you use it in the browser?

**Reply by bumi** [reply]
I am using the library in a browser extension - which basically is the browser - to extract meta data from the current page.

**Reply by BetaHuhn** [reply]
If we can get it to work in the browser and on the server, I am all for it! It just depends on how complicated it will get to support both.

Replacing `got` with a library that works on both environments would be easy, we can just switch to `axios`. How would you replace `domino`?

**Reply by bumi** [reply]
In the browser domino would not be needed as the browser document has native `querySelectorAll` support. 
I just don't know how to switch between domino and the native document.

**Reply by BetaHuhn** [reply]
> I just don't know how to switch between domino and the native document.

Yeah, that's what I meant :) 

We would probably need two separate exports of the package, something like `metadata-scraper/node` and `metadata-scraper/browser` which would share a common core. I'll look into it.

**Reply by rajorpratyush** [reply]
Any plans for supporting browser?

**Reply by BetaHuhn** [reply]
I don’t have time to work on this myself, but a PR is welcome!

My only requirement is that it should still work on the server.
